matter_server/client/client.py

```diff
@@ -338,8 +341,25 @@ async def node_diagnostics(self, node_id: int) -> NodeDiagnostics:
         ):
             if isinstance(last_network_id, bytes):
                 network_name = last_network_id.decode("utf-8")
-            else:
+            elif last_network_id != NullValue:
                 network_name = last_network_id
+        # last resort to get the (wifi) networkname;
+        # enumerate networks on the NetworkCommissioning cluster
+        networks: list[Clusters.NetworkCommissioning.Structs.NetworkInfoStruct]
+        if not network_name and (
+            networks := node.get_attribute_value(
+                0,
+                cluster=None,
+                attribute=Clusters.NetworkCommissioning.Attributes.Networks,
+            )
+        ):
+            for network in networks:
+                if isinstance(network.networkID, bytes):
+                    network_name = network.networkID.decode("utf-8")
```

Comment:
So we just pick the first network from the `NetworkCommissioning` cluster? That could be any WiFi network no? I think this could lead to more problems/confusion then being helpful.

Reply:
nope, it is a network that has been used to commisission the device. So 99% sure this network is active on the device.

FYI: From the devices I have tested;

- some only had the network in the lastNetworkID
- Some only had the network in this list

But it was never the wrong network

Reply:
It is not ANY network. There is another clustercommand to let the device do an active scan for networks and return those

Reply:
Ah I see those are possible options. I see there is `Connected` property, we could use that to make the best possible guess?

Reply:
good suggestion, I'll add that

Reply:
done